test/integration/listener_lds_integration_test.cc

```diff
@@ -409,5 +411,103 @@ TEST_P(ListenerIntegrationTest, MultipleLdsUpdatesSharingListenSocketFactory) {
   }
 }
 
+class RebalancerTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                       public BaseIntegrationTest {
+public:
+  RebalancerTest()
+      : BaseIntegrationTest(GetParam(), ConfigHelper::baseConfig() + R"EOF(
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.tcp_proxy
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
+          stat_prefix: tcp_stats
+          cluster: cluster_0
+)EOF") {}
+
+  void initialize() override {
+    config_helper_.renameListener("tcp");
+    config_helper_.addConfigModifier(
+        [&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) -> void {
+          auto& src_listener_config = *bootstrap.mutable_static_resources()->mutable_listeners(0);
+          src_listener_config.mutable_use_original_dst()->set_value(true);
+          // Note that the below original_dst is replaced by FakeOriginalDstListenerFilter at the
+          // link time.
+          src_listener_config.add_listener_filters()->set_name(
+              "envoy.filters.listener.original_dst");
+          auto& dst_listener_config = *bootstrap.mutable_static_resources()->add_listeners();
+          dst_listener_config = src_listener_config;
+          dst_listener_config.mutable_use_original_dst()->set_value(false);
+          dst_listener_config.clear_listener_filters();
+          dst_listener_config.mutable_bind_to_port()->set_value(false);
+          dst_listener_config.set_name("balanced_target_listener");
+          dst_listener_config.mutable_connection_balance_config()->mutable_exact_balance();
+
+          // 127.0.0.2 is defined in FakeOriginalDstListenerFilter.
+          *dst_listener_config.mutable_address()->mutable_socket_address()->mutable_address() =
```

Comment:
This will fail on ipv6-only. See other tests for how to properly handle this.

Reply:
It won't fail. `bind_to_port=false` listener doesn't use an os socket.